Lib/test/test_unicode.py

```diff
@@ -2819,6 +2819,16 @@ def test_asucs4(self):
             self.assertEqual(unicode_asucs4(s, len(s), 1), s+'\0')
             self.assertEqual(unicode_asucs4(s, len(s), 0), s+'\uffff')
 
+    # Test PyUnicode_AsUTF8()
+    @support.cpython_only
+    def test_asutf8(self):
+        from _testcapi import unicode_asutf8
+
+        self.assertEqual(unicode_asutf8('abc'), 'abc')
+        self.assertEqual(unicode_asutf8('abc\0'), 'abc')
+        self.assertEqual(unicode_asutf8('abc\0abc'), 'abc')
```

Comment:
It's more like you are testing `decodeUTF8`. I'd suggest add more encode samples, BMP, non-BMP, just take from other tests.

Reply:
np, xiang core.